projects/tutorials/orbslam_pointnav_ithor_rgb_ddppo.py

```diff
@@ -0,0 +1,356 @@
+    ]
+
+    ENV_ARGS = dict(
+        width=CAMERA_WIDTH,
+        height=CAMERA_HEIGHT,
+        rotateStepDegrees=10.0,
+        visibilityDistance=1.0,
+        gridSize=0.2,
+        renderDepthImage = True
+    )
+
+    @classmethod
+    def tag(cls):
+        return "PointNavithorRGBPPO"
+
+    @classmethod
+    def training_pipeline(cls, **kwargs):
+        ppo_steps = int(250000000)
+        lr = 3e-4
+        num_mini_batch = 1
+        update_repeats = 3
+        num_steps = 30
+        save_interval = 5000000
+        log_interval = 10000
+        gamma = 0.99
+        use_gae = True
+        gae_lambda = 0.95
+        max_grad_norm = 0.5
+        return TrainingPipeline(
+            save_interval=save_interval,
+            metric_accumulate_interval=log_interval,
+            optimizer_builder=Builder(optim.Adam, dict(lr=lr)),
+            num_mini_batch=num_mini_batch,
+            update_repeats=update_repeats,
+            max_grad_norm=max_grad_norm,
+            num_steps=num_steps,
+            named_losses={"ppo_loss": PPO(**PPOConfig)},
+            gamma=gamma,
+            use_gae=use_gae,
+            gae_lambda=gae_lambda,
+            advance_scene_rollout_period=cls.ADVANCE_SCENE_ROLLOUT_PERIOD,
+            pipeline_stages=[
+                PipelineStage(loss_names=["ppo_loss"], max_stage_steps=ppo_steps)
+            ],
+            lr_scheduler_builder=Builder(
+                LambdaLR, {"lr_lambda": LinearDecay(steps=ppo_steps)}
+            ),
+        )
+
+    def split_num_processes(self, ndevices):
+        assert self.NUM_PROCESSES >= ndevices, "NUM_PROCESSES {} < ndevices {}".format(
+            self.NUM_PROCESSES, ndevices
+        )
+        res = [0] * ndevices
+        for it in range(self.NUM_PROCESSES):
+            res[it % ndevices] += 1
+        return res
+
+    def machine_params(self, mode="train", **kwargs):
+        sampler_devices: Sequence[int] = []
+        if mode == "train":
+            workers_per_device = 1
+            gpu_ids = (
+                []
+                if not torch.cuda.is_available()
+                else self.TRAINING_GPUS * workers_per_device
+            )
+            nprocesses = (
+                1
+                if not torch.cuda.is_available()
+                else self.split_num_processes(len(gpu_ids))
+            )
+            sampler_devices = self.TRAINING_GPUS
+        elif mode == "valid":
+            nprocesses = 1
+            gpu_ids = [] if not torch.cuda.is_available() else self.VALIDATION_GPUS
+        elif mode == "test":
+            nprocesses = 1
+            gpu_ids = [] if not torch.cuda.is_available() else self.TESTING_GPUS
+        else:
+            raise NotImplementedError("mode must be 'train', 'valid', or 'test'.")
+
+        # Disable parallelization for validation process
+        if mode == "valid":
+            for prep in self.PREPROCESSORS:
+                prep.kwargs["parallel"] = False
+
+        observation_set = (
```

Comment:
`ObservationSet` has been deprecated in the latest master (use `SensorPreprocessorGraph` instead).